### ISSUE TITLE
Run external program from within a test

### ISSUE THREAD

**Issue by bartekpacia** [leancodepl/patrol#606]
@mateuszwojtczak mentioned that one of our _big_ customers has complex tests and sometimes needs to run external scripts, e.g to perform the request to the database.

> We've gained the ability to perform actions on behalf of the host from within the test in #593.

We need to think of the API that our users will use for running programs on the host machine.

Here's my initial proposal for the `HostAutomator.runScript()` method:
```dart
Future<void> runScript(String programName, List<String> args]);
```

Example usage:
```dart
// integration_test/database_test.dart

await $.host.runScript('bash', ['./create_user.sh', '--username', 'user', '--some-other-param']);

```

For consideration:
- do we want to resemble `dart:io`'s `Process.run()` and `Process.start()`?
- how will we share stdout and stderr with the user (possibly in real-time with the `start`-variant)?
- ???  please chip in :)

cc: @jBorkowska, @mateuszwojtczak, @jakubfijalkowski, @shilangyu, @fylyppo

**Comment by shilangyu** [comment]
Personally not a fan of the name `runScript` (the `script` part) since it can be used for any process, really.

> how will we share stdout and stderr with the user (possibly in real-time with the start-variant)?

What is the scope of this feature? On one hand I feel like exposing too many features would just encourage usage of it (and I am not sure if it is something worth encouraging), but on the other hand lack of feature might be a blocker for some.

**Comment by bartekpacia** [comment]
True, I agree it's not a good name. What about `runProcess` (and, possibly an async equivalent, `startProcess`)?

> What is the scope of this feature?

Haha, good question! I hope to gather some info from our users: @jBorkowska, @fylyppo, @Kendru98, @zoskar, @lmlikota  – what do you want? :)

**Comment by bartekpacia** [comment]
This has been implemented in #630.
